esp_idf_monitor/base/line_matcher.py

```diff
@@ -23,7 +23,7 @@ class LineMatcher(object):
     def __init__(self, print_filter):
         # type: (str) -> None
         self._dict = dict()
-        self._re = re.compile(r'^(?:\033\[[01];?[0-9]+m?)?([EWIDV]) \([0-9]+\) ([^:]+): ')
+        self._re = re.compile(r'^(?:\033\[[01];?[0-9]+m?)?([EWIDV]) (?:\([0-9]+\)|\([0-9]{2}:[0-9]{2}:[0-9]{2}\.[0-9]{3}\)) ([^:]+): ')
```

Comment:
I would recommend replacing all `[0-9]` with `\d` to simplify the regex.

As mentioned in the issue this only helps with one of the formats and some new formats have been added to ESP-IDF. If you are going to implement those, it would be nice if we could also include some comments because the regex will get much more complicated. If you divide it into multiple lines, you can add a comment per line, so the following structure would be much more readable and maintainable IMO:
```suggestion
        self._re = re.compile(
            r'^(?:\033\[[01];?\d+m?)?'  # ANSI color
            r'([EWIDV] )?'  # optional log level
            r'(?:\(\d+\)|' # ...
            r'\(\d{2}:\d{2}:\d{2}\.\d{3}\)) '  # ...
            r'([^:]+): '  # tag
        )
```
AFAIK to cover the Log v2 we only need to add support for missing timestamps (make it optional and move space inside) and add an optional date to your new system timestamp.

Reply:
Hi @spahijakov, are you still planning to work on this, or should I take over? Thank you!

Reply:
Yeah, please take it up, I was actually looking up how to reconcile bitfield bytestuffing and esp-idf handling of UB misaligned access, on some other project and got sidetracked.
But, yes, @peterdragun we would all like to see this in the upstream - looking forward!